packages/cli/src/commands/update.ts

```diff
@@ -0,0 +1,176 @@
+            if (error) {
+                reject(error);
+            } else {
+                const installedPackages = JSON.parse(stdout.toString());
+                switch (manager) {
+                    case "npm":
+                        resolve(installedPackages);
+                        break;
+                    case "yarn":
+                        resolve(convertYarnOutput(installedPackages));
+                        break;
+                    case "pnpm":
+                        resolve(convertPnpmOutput(installedPackages));
+                        break;
+                }
+            }
+        });
+    });
+}
+
+function convertPnpmOutput(pnpmOutput: any[]): any {
+    const rootProject = pnpmOutput[0];
+    if (rootProject && rootProject.dependencies) {
+        const dependencies: { [name: string]: { version: string } } = {};
+        for (const [name, details] of Object.entries(rootProject.dependencies)) {
+            dependencies[name] = { version: (details as any).version };
+        }
+        return { dependencies };
+    }
+
+    return {};
+}
+
+function convertYarnOutput(yarnOutput: any): any {
+    const packages: { [name: string]: { version: string } } = {};
+
+    for (const pkg of yarnOutput.data.trees) {
+        if (pkg.name) {
+            const [name, version] = pkg.name.split('@');
+            if (name && version) {
+                packages[name] = { version };
+            }
+        }
+    }
+
+    return { dependencies: packages };
+}
+
+async function getLatestVersions(packageNames: string[], packageVersion: { [name: string]: string}): Promise<{ [packageName: string]: string }> {
+    const registryUrl = 'https://registry.npmjs.org';
+
+    const requests = packageNames.map(packageName =>
+        fetch(`${registryUrl}/${packageName}`, {
+            method: "GET",
+            headers: {
+                'Accept-Encoding': 'application/json',
+            },
+        })
+    );
+
+    const responses = await Promise.all(requests);
+    const responseData: any[] = await Promise.all(responses.map(res => res.json()));
+
+    return responseData.reduce((versions: { [packageName: string]: string }, response, index) => {
+        const packageName = packageNames[index];
+        if (packageName) {
+            const isUsingNext = packageVersion[packageName]?.includes("next");
+            const nextVersion = response['dist-tags']?.next;
+            const latestVersion = response['dist-tags'].latest;
+
+            if (isUsingNext && nextVersion && semver.gt(nextVersion, latestVersion)) {
+                versions[packageName] = nextVersion;
+            } else {
+                versions[packageName] = latestVersion;
+            }
+        }
+        return versions;
+    }, {});
+}
+
+function updatePackageJson(updates: { [packageName: string]: string }) {
+    logger.info(JSON.stringify(updates));
+    const packageJson = JSON.parse(readFileSync('package.json', 'utf8'));
```

Comment:
This reading has two issues, one the package.json can be incorrect and needs to be built using the resolvedPath. The other is we already have `readJSONFile` and `readJSONFileSync` utils, so please use one of those.

Reply:
done